### PR TITLE
Stand up Bot Coordinator Azure Function 

### PR DESCRIPTION
**Pull request by NickGerleman**
This change adds a new package for the bot coordinator, adapting a template function app to engineering systems in our repo. Right now it does very little, and it mostly just structure and deployment.

Deployment of the function app is tied to nightly publish, partially for ease of use, but also to ensure changes to monorepo packages are reflected. Azure credentials are tied as a service connection to Pipelines, only for explicitly authorized pipelines (i.e. publish).

Validated building/running the function app locally. Have tested bits of the deploy, but not everything end to end. That might need some iteration based on how Azure Pipelines likes it, but its job is independent of the rest of publish so it shouldn't degrade it (apart from potential YAML parse errors).

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6900)

### CODE REVIEW

**Review by asklar — CHANGES_REQUESTED** [review]
It's not clear to me what this would be used for or by whom, and it's saying a new development dependency so I'd like to understand the intent of this with better before I sign off
Is there a short design doc/speclet I can read to understand this intent and what this would achieve?

**Comment by NickGerleman** [comment]
> It's not clear to me what this would be used for or by whom, and it's saying a new development dependency so I'd like to understand the intent of this with better before I sign off
> Is there a short design doc/speclet I can read to understand this intent and what this would achieve?

There is a dev doc that was distributed over Teams and to the ReactWindowsCoreTeam DL last week. There's a PR to add the document to the repo as well, but it lacks some of the comments on the doc, or some internal bits. 

TLDR is that we want to set up bots that need to respond to webhooks. That requires running a light service. We have funding, etc to do that.

The development dependency is needed if wanting to test the function app locally. It's pretty light so it's convenient to have the script install it along with others even if most won't encounter it.

**Review by NikoAri — APPROVED** [review]
:shipit:

**Review by asklar — APPROVED** [review]
see if we can reuse the tool that virtual environments would install

**Comment by NickGerleman** [comment]
@asklar for now the dependency should only be needed for local validation, and not in CI. Might change in the future if I figure out a nice way to do E2E validation of the whole function app (for now I'm looking at some more scoped tests).
